ImageButtonHandler, AnimationManager, Fix Tests, CI...

drasticactions opened the pull request:
This ended up being a bigger PR than I thought.

- Moves ImageButtonHandler/MauiImageButton into the new locations and refactors them.
- Adds new tests for them
- Adds an updates ControlGallery entry for them.

- Add MauiPlanets demo app. It should work except for the CollectionView images in the gallery not loading. That needs more work.
- But that led to finding out that we need to also implement the AnimationManager to get proper animations on the UI Thread for Avalonia. I did that and have tests for it...

- Which then led me to find out that tests were failing in CI and ended up green, because I am dumb and set it to Continue On Fail, thinking that posting the results would cause it to fail, which it doesn't. I need to let it still fail. The results will post after.

I then fixed the failing tests, which were either handles not being created for stubs, and the use of Reflection for the ButtonHandler tests for getting the content. We should never use reflection in the standard library, so I put skip on those tests until we can come up with a better solution.

And I turned on Treating Warnings as Failures for Avalonia.Controls.Maui and Avalonia.Controls.Maui.Tests, fixes #56.

drasticactions committed:
Start ImageButton Handler move

drasticactions committed:
Add Command tests

drasticactions committed:
Update samples

drasticactions committed:
Implement Animations

drasticactions committed:
Fail tests on error

drasticactions committed:
Fix test pass one

drasticactions committed:
Skip test

drasticactions committed:
Readd handlers

drasticactions committed:
Add simple tests

drasticactions committed:
Merge branch 'main' into image-button-handler

# Conflicts:
#	samples/ControlGallery/ControlGallery/Pages/ButtonPage.xaml.cs

drasticactions committed:
Update

drasticactions committed:
Turn on warnings as errors

drasticactions committed:
Remove Reflection on Generic Type

jsuarezruiz commented on samples/ControlGallery/ControlGallery/Pages/ButtonPage.xaml:
```diff
@@ -476,7 +476,471 @@
-            
+
+            <!-- ImageButton Section -->
+            <Label Text="ImageButton Control"
```
Could we separate in a different Page and include a new `TextCell` in the MainPage?

drasticactions replied:
Fixed with https://github.com/AvaloniaUI/Avalonia.Controls.Maui/pull/60/commits/513049aa6860830c824f67c0e74f25f5d4fb8490

jsuarezruiz commented on samples/ControlGallery/ControlGallery/Pages/ButtonPage.xaml:
```diff
@@ -476,7 +476,471 @@
+                            <ImageButton Source="dotnet_bot.png"
+                                         WidthRequest="70"
+                                         HeightRequest="70"
+                                         Command="{Binding ImageButtonParameterizedCommand}"
```
Cannot see updates in the Label? Can review it?

drasticactions replied:
Fixed with https://github.com/AvaloniaUI/Avalonia.Controls.Maui/pull/60/commits/911a1c9851ef4ec63fd19318fbfcb7d806e12cdf

jsuarezruiz reviewed (changes requested):
Should we create a Samples (or similar name: Showcase, etc) folder to group projects with samples?

There is a clipping issue in the profile page:
<img width="1129" height="642" alt="Image" src="https://github.com/user-attachments/assets/02c55b95-b1d5-4cfe-9867-2dbcb7a8893e" />
Will not pay a lot of attention to it because use a Border and is implemented but we need to review it and complete it (refactor handler, add tests etc).

Also the fake tabbar is on top of the planets lists (can fix easily in the sample).

Then, in details, the Gallery not display anything:

<img width="927" height="626" alt="Image" src="https://github.com/user-attachments/assets/e2f5201a-c783-4562-b146-b6e520bc0396" />
Use a CollectionView and could happen something similar to the Border. 

Once we work around Border and CollectionView, this sample should work nicely.

drasticactions committed:
Move MauiPlanets

drasticactions committed:
Refactor ControlGallery to remove unneeded resources

drasticactions committed:
Migrate ImageButton sample

drasticactions committed:
Fix warnings

drasticactions committed:
Fix command

drasticactions commented:
@jsuarezruiz I moved MauiPlanets to a `showcase` directory, and refactored it to removing the build warnings and update the APIs.The images should now be working.

<img width="675" height="1018" alt="スクリーンショット 2025-12-06 10 19 45" src="https://github.com/user-attachments/assets/456b9e22-4022-4ee6-a4f0-623fadbd0d99" />

I also refactored ControlGallery and MauiPlanets to get all of the resources into the class library and reference them from there, cutting down on the duplicated files.

drasticactions committed:
Fix race condition

drasticactions committed:
Apply clip to image

drasticactions commented:
@jsuarezruiz 
<img width="103" height="109" alt="スクリーンショット 2025-12-06 11 10 47" src="https://github.com/user-attachments/assets/ec7af0c5-352e-4c35-98c5-9e8193d56dd7" />

https://github.com/AvaloniaUI/Avalonia.Controls.Maui/pull/60/commits/f713af2bc60574fc06ed55aaaf530cb8b111920c

Applying the clip to Image should fix that.

drasticactions committed:
Bring back README and STATUS
